src/osiris_log.erl

```diff
@@ -2009,6 +2038,18 @@ overview(Dir) ->
             {Range, EpochOffsets}
     end.
 
+index_files_with_segment([<<_:20/binary, ".segment">> | Rem], Dir, Acc) ->
+    %% orphaned segment file, ignore
+    index_files_with_segment(Rem, Dir, Acc);
+index_files_with_segment([<<_:20/binary, ".index">> = I,
+                          <<_:20/binary, ".segment">>
```

Comment:
Do you need to match parts before the file extension like you do in `orphaned_segments/2`?

Or perhaps call `orphaned_segments/1` to avoid a duplicated logic?

Reply:
yes! good catch!